src/main/java/com/stripe/model/EventDataObjectDeserializer.java

```diff
@@ -185,13 +182,14 @@ public StripeObject deserializeUnsafeWith(CompatibilityTransformer transformer)
         transformer.transform(rawJsonObject.deepCopy(), apiVersion, eventType));
   }
 
+  @SuppressWarnings("StringSplitter")
   private boolean apiVersionMatch() {
-    return getIntegrationApiVersion().equals(this.apiVersion);
-  }
+    // Test that only the actual API version component of the version strings match. The integration
+    // API version might have beta headers that are not present in the event payload.
+    String integrationApiVersionStableComponent = Stripe.stripeVersion.split(";")[0];
+    String payloadApiVersionStableComponent = this.apiVersion.split(";")[0];
```

Comment:
Is this required?  I haven't done it in my PRs.

Reply:
You're right, I don't think so. Updated.